plugin/gcp/keypair.py

```diff
@@ -0,0 +1,132 @@
+# #######
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+import os
+
+from Crypto.PublicKey import RSA
+
+from plugin.gcp.service import GoogleCloudPlatform
+from plugin.gcp.service import blocking
+from plugin.gcp.service import GCPError
+from plugin.gcp import utils
+
+
+class KeyPair(GoogleCloudPlatform):
+    KEY_NAME = 'key'
+    KEY_VALUE = 'sshKeys'
+
+    def __init__(self,
+                 config,
+                 logger,
+                 user,
+                 private_key_path):
+        """
+        Create KeyPair object
+
+        :param config: gcp auth file
+        :param logger: logger object
+        :param user: name of the user to authenticate
+        :param private_key_path: path where private key is stored
+        """
+        super(KeyPair, self).__init__(config, logger)
```

Comment:
here also I prefer passing the user and private_key_path args directly to the methods that require them and not to init